### PR TITLE
chore: GitHub Actions integration

### PR DESCRIPTION
**Pull request by robertwe**
## Change description
- GitHub Actions integration

### CODE REVIEW

**Comment by ayarotsky** [comment]
@robertwe Please enable GHA for the repo.

**Review by urkle — APPROVED** [review]

**Review by urkle — CHANGES_REQUESTED** [review]
@robertwe I force pushed it back up to trigger it to run the GH Actions and it failed with an unknown job rubocop error.

**Comment by robertwe** [comment]
> @robertwe I force pushed it back up to trigger it to run the GH Actions and it failed with an unknown job rubocop error.

Noticed that already and it should be fixed now.

**Review by urkle — APPROVED** [review]

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+      - uses: actions/cache@v2
+        with:
+          path: $HOME/.cache/rubocop_cache
+          key: process-balancer-rubocop-${{ hashFiles('.rubocop.yml') }}
+          restore-keys: |
+            process-balancer-rubocop-${{ hashFiles('.rubocop.yml') }}
+            process-balancer-rubocop-
+
+      - name: Rubocop
```
Comment (ayarotsky) [inline .github/workflows/main.yml:59]: Should we extract rubocop into a separate parallel step as we did in all other pipelines?

Comment (robertwe) [reply .github/workflows/main.yml:59]: Of course we can do it but in this case I do not see much benefits. Rubocop step takes ~4[s] please take a look here: https://github.com/NetsoftHoldings/process_balancer/runs/4285697235?check_suite_focus=true

**Review by ayarotsky — APPROVED** [review]
🚀

**Pull request merged**